packages/ui/src/services/mapping.service.test.ts

```diff
@@ -66,6 +66,28 @@ describe('MappingService', () => {
       MappingService.removeAllMappingsForDocument(tree, DocumentType.PARAM, 'sourceParam1');
       expect(tree.children.length).toEqual(1);
     });
+
+    it('should remove mappings related to the removed param in case of XML schema', () => {
+      targetDoc = TestUtil.createTargetOrderDoc();
+      paramsMap = TestUtil.createParameterMap();
+      tree = new MappingTree(targetDoc.documentType, targetDoc.documentId);
+      MappingSerializerService.deserialize(conditionalMappingsToShipOrderXslt, targetDoc, tree, paramsMap);
+
+      expect(tree.children.length).toEqual(1);
+      MappingService.removeAllMappingsForDocument(tree, DocumentType.PARAM, 'cart');
+      expect(tree.children.length).toEqual(0);
+    });
+
+    it('should remove mappings related to the removed param in case of JSON schema', () => {
+      const targetJSONDoc = TestUtil.createJSONTargetOrderDoc();
+      paramsMap = TestUtil.createJSONParameterMap();
+      tree = new MappingTree(targetJSONDoc.documentType, targetJSONDoc.documentId);
+      MappingSerializerService.deserialize(conditionalMappingsToShipOrderJsonXslt, targetJSONDoc, tree, paramsMap);
+
+      expect(tree.children.length).toEqual(1);
+      MappingService.removeAllMappingsForDocument(tree, DocumentType.PARAM, 'cart-x');
+      expect(tree.children[0].children.length).toEqual(0);
```

Comment:
Sorry if I forgot we already talked, isn't this supposed to be `tree.children.length` to be 0 instead? Or in case it doesn't yet work as expected due to the other bug, it might be better to use `it.skip` and let it skip the execution until the bug is fixed, and put a comment with the link to the issue. Then we can keep the expectations to be what we really want.